### PR TITLE
feat!: ignore queryparams in matching

### LINKED ISSUE #54: Is there a way to ignore query parameters?
### Environment

Node version: 18.15.0
Package version: 1.0.1

### Reproduction

```ts
import { createRouter } from 'radix3';

const routes = ['/', '/login'];

const skipRoutes = createRouter({ strictTrailingSlash: false });
routes.forEach((e) => skipRoutes.insert(e, {}));

const result = skipRoutes.lookup('/login?foo=1234&bar=1234');

console.log(result); // null
```

### Describe the bug

The `lookup` method doesn't match routes with query params with the ones that don't have query params.

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by kingtimm**
### 🔗 Linked issue

fixes #54 - inclusion of queryparameters on a path make it fail to match.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Removed query parameters from the path before it is matched. The router did not seem to support them anyway.

My use-case was in nuxt-vitest's registerEndpoint but found others running into it.

Resolves #54 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/matcher.ts [hunk 1]
```diff
@@ -85,6 +85,9 @@ function _matchRoutes(path: string, table: RouteTable): RadixNodeData[] {
   // Order should be from less specific to most specific
   const matches = [];
 
+  // Ignore query parameters
+  path = path.split("?")[0];
```
Comment (pi0) [inline src/matcher.ts:89]: We need to use index check (like https://github.com/unjs/h3/blob/main/src/router.ts#L86) to handle URL like `/login?returnTo=/path?foo=bar` (context: https://github.com/unjs/h3/issues/190)

**Comment by pi0** [comment]
Thanks for PR dear @kingtimm and sorry it is delayed. Main reason is that, the consumer libs (mainly unjs/h3) are already doing this normalization. It is not a breaking-change per-se but doing this means performance overhead of iterating string twice. Because of this i plan this in next major release to be included so we can adjust h3 router behavior while upgrading.

**Comment by mukundshah** [comment]
I suggest this should be optional. I some case, we may need to match the route along with the query params;

Maybe having something like `{query: true}` and `false` being the default value, we could have this without breaking anything.

**Comment by pi0** [comment]
Hi dear @kingtimm. I had been thinking about this for a while and while in some direct usecases it makes sense, i think it would be the best to let it up to the users to remove query params and pass pathname themselves.

In cases of using with web APIs, if users already parsed URL with `new URL`, they already have access to `url.pathname`. This improves perf, reusability (if we drop internally in rou3, users need to parse again) and also stability since URL parsing has many edge cases.

**Pull request closed without merging**